netebpfext/net_ebpf_ext_bind.c

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+/*++
+
+Abstract:
+
+   This file implements the BIND program type hook on eBPF for Windows.
+
+Environment:
+
+    Kernel mode
+
+--*/
+#define INITGUID
+
+#include "net_ebpf_ext.h"
+
+// ebpf_bind_program_data.h has generated
+// headers. encode_program_info generates them from the structs
+// in ebpf_nethooks.h. This workaround exists due to the inability
+// to call RPC serialization services from kernel mode. Once we switch
+// to a different serializer, we can get rid of this workaround.
+#include "ebpf_bind_program_data.h"
+
+static ebpf_ext_attach_hook_provider_registration_t* _ebpf_bind_hook_provider_registration = NULL;
+static ebpf_extension_provider_t* _ebpf_bind_program_info_provider = NULL;
+
+static ebpf_context_descriptor_t _ebpf_bind_context_descriptor = {
+    sizeof(bind_md_t), EBPF_OFFSET_OF(bind_md_t, app_id_start), EBPF_OFFSET_OF(bind_md_t, app_id_end), -1};
+static ebpf_program_info_t _ebpf_bind_program_info = {{"bind", &_ebpf_bind_context_descriptor, {0}}, 0, NULL};
+
+static ebpf_program_data_t _ebpf_bind_program_data = {&_ebpf_bind_program_info, NULL};
+
+static ebpf_extension_data_t _ebpf_bind_program_info_provider_data = {
+    NET_EBPF_EXTENSION_NPI_PROVIDER_VERSION, sizeof(_ebpf_bind_program_data), &_ebpf_bind_program_data};
+
+// b9707e04-8127-4c72-833e-05b1fb439496
+DEFINE_GUID(EBPF_ATTACH_TYPE_BIND, 0xb9707e04, 0x8127, 0x4c72, 0x83, 0x3e, 0x05, 0xb1, 0xfb, 0x43, 0x94, 0x96);
+
+DEFINE_GUID(EBPF_PROGRAM_TYPE_BIND, 0x608c517c, 0x6c52, 0x4a26, 0xb6, 0x77, 0xbb, 0x1c, 0x34, 0x42, 0x5a, 0xdf);
+
+static void
+_net_ebpf_ext_resource_truncate_appid(bind_md_t* ctx)
+{
+    wchar_t* last_separator = (wchar_t*)ctx->app_id_start;
+    for (wchar_t* position = (wchar_t*)ctx->app_id_start; position < (wchar_t*)ctx->app_id_end; position++) {
+        if (*position == '\\') {
+            last_separator = position;
+        }
+    }
+    if (*last_separator == '\\') {
+        last_separator++;
+    }
+    ctx->app_id_start = (uint8_t*)last_separator;
+}
+
+void
+net_ebpf_ext_resource_allocation_classify(
+    _In_ const FWPS_INCOMING_VALUES* incoming_fixed_values,
+    _In_ const FWPS_INCOMING_METADATA_VALUES* incoming_metadata_values,
+    _Inout_opt_ void* layer_data,
+    _In_opt_ const void* classify_context,
+    _In_ const FWPS_FILTER* filter,
+    uint64_t flow_context,
+    _Inout_ FWPS_CLASSIFY_OUT* classify_output)
+/* ++
+
+   A simple classify function at the WFP Resource Allocation layer.
+
+-- */
```

Comment:
replace with doxygen style comment.  Same for line 129 etc.  doxygen comments always go above the prototype, not below.